From e4a83c942a1e232582ec9288cb280d866713b425 Mon Sep 17 00:00:00 2001
From: Rastrisr <rastrisr@tutamail.com>
Date: Thu, 15 Oct 2026 01:58:58 +0000
Subject: [PATCH] LineageSettings: add Global.GLOBAL_VPN_APP

Needed by CalyxOS Global VPN (fw/b 7cad176d, Settings 94b7f5bc); equivalent of calyx-sdk change 7540.

Change-Id: Iff8c07c4a25a363b9d217b1ce74a901fdf9129f6
---
 sdk/src/java/lineageos/providers/LineageSettings.java | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/sdk/src/java/lineageos/providers/LineageSettings.java b/sdk/src/java/lineageos/providers/LineageSettings.java
index 9fe5cb4..2a845bc 100644
--- a/sdk/src/java/lineageos/providers/LineageSettings.java
+++ b/sdk/src/java/lineageos/providers/LineageSettings.java
@@ -3417,6 +3417,16 @@ public final class LineageSettings {
         /** @hide */
         public static final Validator TRUST_RESTRICT_USB_VALIDATOR =
                 new InclusiveIntegerRangeValidator(0, 2);
+
+        /**
+         * Package name of the VPN app whose connection carries traffic for all users and
+         * profiles ("Global VPN", from CalyxOS). Empty/unset = no global VPN.
+         * @hide
+         */
+        public static final String GLOBAL_VPN_APP = "global_vpn_app";
+
+        /** @hide */
+        public static final Validator GLOBAL_VPN_APP_VALIDATOR = sAlwaysTrueValidator;
         // endregion
 
         /**
@@ -3447,6 +3457,7 @@ public final class LineageSettings {
                 new ArrayMap<String, Validator>();
         static {
             VALIDATORS.put(TRUST_RESTRICT_USB, TRUST_RESTRICT_USB_VALIDATOR);
+            VALIDATORS.put(GLOBAL_VPN_APP, GLOBAL_VPN_APP_VALIDATOR);
             VALIDATORS.put(__MAGICAL_TEST_PASSING_ENABLER,
                     __MAGICAL_TEST_PASSING_ENABLER_VALIDATOR);
         };
-- 
2.43.0

